libraries/ts/src/sbv2.ts

```diff
@@ -1168,16 +1149,16 @@ export class AggregatorAccount {
       params.authority ?? this.keypair ?? programWallet(this.program);
     return program.methods
       .aggregatorSetConfig({
-        name: (params.name ?? Buffer.from("")).slice(0, 32),
-        metadata: (params.metadata ?? Buffer.from("")).slice(0, 128),
-        batchSize: params.batchSize,
-        minOracleResults: params.minOracleResults,
-        minUpdateDelaySeconds: params.minUpdateDelaySeconds,
-        minJobResults: params.minJobResults,
-        forceReportPeriod: new anchor.BN(params.forceReportPeriod ?? 0),
-        varianceThreshold: params.varianceThreshold
-          ? SwitchboardDecimal.fromBig(new Big(params.varianceThreshold))
-          : undefined,
+        name: findKeyInObject(params, "name"),
+        metadata: findKeyInObject(params, "metadata"),
+        batchSize: findKeyInObject(params, "batchSize"),
+        minOracleResults: findKeyInObject(params, "minOracleResults"),
+        minUpdateDelaySeconds: findKeyInObject(params, "minUpdateDelaySeconds"),
+        minJobResults: findKeyInObject(params, "minJobResults"),
+        forceReportPeriod: findKeyInObject(params, "forceReportPeriod"),
+        varianceThreshold: findKeyInObject(params, "varianceThreshold", (o) =>
```

Comment:
Will undefined or defaults actually work here? I thought the anchor client didn't infer these sized buffers (though they do move at a crazy pace). If we wanna use this on the client then we probably want them to be optional for min effort

Reply:
All of the parameters in `AggregatorSetConfigParams` are optional. The purpose of `findKeyInObject` is to have a consistent method of searching the parameters for a key, returning null if not found or optionally transforming (from `number` to `SwitchboardDecimal` for example) as needed.